### PR TITLE
Terms of Agreement

### PR DESCRIPTION
**Pull request by miriam-shamash**
Made the user agree to a terms of agreement before signing up for an account as either a student or an alumni.

Screen Shot 2020-07-30 at 6 01 44 PM

### CODE REVIEW

File: client/src/components/Signup.js [hunk 1]
```diff
@@ -271,6 +278,16 @@ export default class Signup extends React.Component {
         this.setState({imageModalOpen: !this.state.imageModalOpen})
     }
 
+    handleTermsModal() {
+        this.setState({termsModalOpen: !this.state.termsModalOpen})
+    }
+
+    handleCheckedAgreement(e, data){
+        console.log("Checked value is", data.checked)
```
Comment (awilli70) [inline client/src/components/Signup.js:286]: remove log

File: client/src/components/Signup.js [hunk 2]
```diff
@@ -858,7 +875,7 @@ export default class Signup extends React.Component {
                         >
                         </Form.Dropdown>
                         <Form.Field
-                            required={true}
+                            //required={true}
```
Comment (awilli70) [inline client/src/components/Signup.js:878]: I'm pretty sure required can be uncommented - all it does is add the red asterisk next to the field label

File: client/src/components/Signup.js [hunk 3]
```diff
@@ -887,6 +904,30 @@ export default class Signup extends React.Component {
...
+                        <Divider/>
+                        <Form.Field
+                            required={true}
+                            style = {fieldStyle}
+                        >
+                            <label> Terms of Agreements</label>
+                            <Button
+                                primary color="blue"
+                                type="button"
+                                onClick={() => {this.setState({termsModalOpen: true})}}
```
Comment (awilli70) [inline client/src/components/Signup.js:916]: This is fine, but the callback may be simplified by having the onClick just take this.handleTermsModal

File: client/src/components/Signup.js [hunk 4]
```diff
@@ -10,6 +10,7 @@ import MajorSelectionModal from './MajorSelectionModal';
 import CompanySelectionModal from './CompanySelectionModal';
 import JobTitleSelectionModal from './JobTitleSelectionModal';
 import ImageSelectModal from './ImageSelectModal';
+import TermsOfAgreementsModal from'./TermsOfAgreementsModal';
```
Comment (reshadbinharun) [inline client/src/components/Signup.js:13]: nit: Lets call it the `TermsOfAgreementModal`

File: client/src/components/Signup.js [hunk 5]
```diff
@@ -319,7 +335,7 @@ export default class Signup extends React.Component {
                     }}
                 />
             </Button>
-            </>
+            </>
```
Comment (reshadbinharun) [inline client/src/components/Signup.js:338]: remove trailing white space

File: client/src/components/Signup.js [hunk 6]
```diff
@@ -887,6 +903,30 @@ export default class Signup extends React.Component {
                             this.getAlumniFields() :
                             this.getStudentFields()
                         }
+                        <Divider/>
+                        <Form.Field
+                            required={true}
+                            style = {fieldStyle}
+                        >
+                            <label> Terms of Agreements</label>
```
Comment (reshadbinharun) [inline client/src/components/Signup.js:911]: Terms of Agreement*

File: client/src/components/Signup.js [hunk 7]
```diff
@@ -887,6 +903,30 @@ export default class Signup extends React.Component {
...
+                                size="mini"
+                            >
+                            View Terms of Agreements
+                            </Button>
+                            <TermsOfAgreementsModal
+                                modalOpen={this.state.termsModalOpen}
+                                close={this.handleTermsModal}
+                            />
+                            <Checkbox 
+                                label = 'I agree to the Terms of Agreements'
```
Comment (reshadbinharun) [inline client/src/components/Signup.js:925]: Agreement*

File: client/src/components/TermsOfAgreementsModal.js [hunk 8]
```diff
@@ -0,0 +1,68 @@
+import React, { Component } from 'react';
+import {Button, Modal } from 'semantic-ui-react';
+
+/*
+    props:
+    - modalOpen
+    - closeModal: ()
+*/
+export default class JobTitleSelectionModal extends Component {
```
Comment (reshadbinharun) [inline client/src/components/TermsOfAgreementsModal.js:9]: class name should be `TermsOfAgreementModal`

File: client/src/components/TermsOfAgreementsModal.js [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+    }
+
+    render() {
+        return (
+            <Modal
+                open={this.props.modalOpen}
+                style={{overflow: 'auto', maxHeight: 600}}
+            >
+                <Modal.Header>Terms Of Agreements</Modal.Header>
+                <Modal.Content>
```
Comment (reshadbinharun) [inline client/src/components/TermsOfAgreementsModal.js:21]: Let's add a `// TODO: Replace with actual terms`

**Comment by reshadbinharun** [comment]
Let's uncommit the .DS_Store as they pertain to your personal mac settings! A couple of comments but this is looking great!

**Comment by miriam-shamash** [comment]
> .DS_Store

e

**Comment by reshadbinharun** [comment]
Let's add some margin to the checkbox, a prop like `style={{ 'margin-top': '5px' }}` should do it!

File: client/src/components/Signup.js [hunk 10]
```diff
@@ -888,6 +904,30 @@ export default class Signup extends React.Component {
                             this.getStudentFields()
                         }
                         <Divider/>
+                        <Form.Field
+                            required={true}
+                            style = {fieldStyle}
+                        >
+                            <label> Terms of Agreement</label>
```
Comment (reshadbinharun) [inline client/src/components/Signup.js:911]: Let's remove this label.

File: client/src/components/TermsOfAgreementModal.js [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+        super(props)
+    }
+
+    render() {
+        return (
+            <Modal
+                open={this.props.modalOpen}
+                style={{overflow: 'auto', maxHeight: 600}}
+            >
+                // TODO: Replace with actual terms
```
Comment (reshadbinharun) [inline client/src/components/TermsOfAgreementModal.js:20]: Move this TODO comment to before the `render() { ...` as this is showing up in the modal
image

**Pull request merged**